github/commitstatus.go

```diff
@@ -96,14 +102,44 @@ func (s CommitStatus) Add(sha, state, targetURL, description string) error {
 	req.Header.Set("Accept", "application/vnd.github.v3+json")
 	req.Header.Set("Content-Type", "application/json")
 
+	for attempt := 0; attempt < maxRetries; attempt++ {
+		s.log.Info(fmt.Sprintf("Http request attempt: %d out of %d", attempt+1, maxRetries))
```

Comment:
Note that instead of bothering with the `+1` and `-1` here and there, you could change the loop:
```suggestion
	for attempt := 1; attempt <= maxRetries; attempt++ {
		s.log.Info(fmt.Sprintf("Http request attempt: %d out of %d", attempt, maxRetries))
```

Reply:
Changed as suggested...